mc-sema/bin_descend/get_cfg.py

```diff
@@ -449,6 +449,54 @@ def isExternalData(fn):
         return False
 
 
+def sanityCheckJumpTableSize(table_ea, ecount):
+    """ IDA doesn't correctly calculate some  jump table sizes """
+    if not isLinkedElf():
+        return ecount
+
+    if getBitness() != 64:
+        return ecount
+
+    table_insn = idautils.DecodeInstruction(table_ea)
+    if table_insn is None:
+        DEBUG("Could not decode instruction at {:x}\n".format(table_insn))
+        return ecount
+
+    if table_insn.Operands[0].type != idc.o_reg:
+        return ecount
+
+    DEBUG("Sanity checking table at {:x}\n".format(table_ea))
+
+    # get register we jump with
+    jmp_reg = table_insn.Operands[0].value
+
+    inst_ea = table_ea
+    # walk back up to 5 instructions
+    for i in xrange(5):
```

Comment:
The cmp (see screenshot above) compares the maximum index value for the jump table. The ideal solution to this problem is to get IDA to fix it upstream, but latest IDA has python API changes also, so the whole fix would be difficult, hence special casing it for now.